h5netcdf/core.py

```diff
@@ -222,10 +222,19 @@ def __init__(self, parent, name):
                         assert dim_id is not None
                         coord_ids = v.attrs['_Netcdf4Coordinates']
                         size = v.shape[list(coord_ids).index(dim_id)]
+                        current_size = size
                     else:
                         assert len(v.shape) == 1
-                        size = v.size
+                        # Unlimited dimensions are represented as None.
+                        if v.maxshape == (None,):
+                            size = None
+                            current_size = v.size
+                        else:
+                            size = v.size
+                            current_size = v.size
```

Comment:
This is the same in both conditions, so it doesn't need to be in the condition. You might also be able to fit the condition on one line, e.g.,
```python
size = None if v.maxshape == (None,) else v.size
current_size = v.size
```